src/main/java/io/split/android/engine/matchers/semver/Semver.java

```diff
@@ -157,6 +155,12 @@ private void setMajorMinorAndPatch(String version) throws SemverParseException {
     private String setVersion() {
         String toReturn = mMajor + VALUE_DELIMITER + mMinor + VALUE_DELIMITER + mPatch;
         if (mPreRelease != null && mPreRelease.length != 0) {
+            for (int i = 0; i < mPreRelease.length; i++) {
+                if (isNumeric(mPreRelease[i])) {
```

Comment:
what happen if the component is a string? are you discarding it ?

Reply:
Not discarding it. `mPreRelease` is already an array of `String`, if it's numeric I replace it with the element with the `Long` version; otherwise I leave it as it was.